src/autopas/containers/linkedCells/traversals/C18Traversal.h

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file C18Traversal.h
+ * @author nguyen
+ * @date 06.09.2018
+ */
+
+#pragma once
+
+#include "LinkedCellTraversalInterface.h"
+#include "autopas/containers/cellPairTraversals/C18BasedTraversal.h"
+#include "autopas/pairwiseFunctors/CellFunctor.h"
+#include "autopas/utils/WrapOpenMP.h"
+
+namespace autopas {
+
+/**
+ * This class provides the c18 traversal.
+ *
+ * The traversal uses the c18 base step performed on every single cell. Since
+ * these steps overlap a domain coloring with eighteen colors is applied.
+ *
+ * @tparam ParticleCell the type of cells
+ * @tparam PairwiseFunctor The functor that defines the interaction of two particles.
+ * @tparam useSoA
+ * @tparam useNewton3
+ */
+template <class ParticleCell, class PairwiseFunctor, bool useSoA, bool useNewton3>
+class C18Traversal : public C18BasedTraversal<ParticleCell, PairwiseFunctor, useSoA, useNewton3>,
+                     public LinkedCellTraversalInterface<ParticleCell> {
+ public:
+  /**
+   * Constructor of the c18 traversal.
+   * @param dims The dimensions of the cellblock, i.e. the number of cells in x,
+   * y and z direction.
+   * @param pairwiseFunctor The functor that defines the interaction of two particles.
+   */
+  explicit C18Traversal(const std::array<unsigned long, 3> &dims, PairwiseFunctor *pairwiseFunctor)
+      : C18BasedTraversal<ParticleCell, PairwiseFunctor, useSoA, useNewton3>(dims, pairwiseFunctor),
+        _cellFunctor(
+            CellFunctor<typename ParticleCell::ParticleType, ParticleCell, PairwiseFunctor, useSoA, useNewton3>(
+                pairwiseFunctor)) {
+    computeOffsets();
+  }
+
+  /**
+   * @copydoc LinkedCellTraversalInterface::traverseCellPairs()
+   */
+  void traverseCellPairs(std::vector<ParticleCell> &cells) override;
+
+  /**
+   * Computes all interactions between the base
+   * cell and adjacent cells with greater a ID.
+   * @param cells vector of all cells.
+   * @param x x of base cell
```

Comment:
x index

Reply:
Also applies to next lines

Reply:
https://github.com/AutoPas/AutoPas/pull/148/commits/64410a25a456134d1c66ddd20243ffd22c1ecbc4